pkg/identity/cache/allocator.go

```diff
@@ -204,6 +222,35 @@ func AllocateIdentity(ctx context.Context, lbls labels.Labels) (*identity.Identi
 		return nil, false, err
 	}
 
+	identityRefCountMutex.Lock()
+	refCountNew := idPoolRefCount[id] == 0
+	if refCountNew {
+		identityControllerManager.UpdateController(fmt.Sprintf("sync-identity (%d)", id),
+			controller.ControllerParams{
+				DoFunc: func(ctx context.Context) error {
+					// We just allocated the identity a couple lines above,
+					// when a controller is added / updated, it starts
+					// immediately, to avoid re-allocating the recently identity
+					// we will sleep for 5 minutes
+					t := time.NewTicker(5 * time.Minute)
+					defer t.Stop()
+					select {
+					case <-t.C:
+					case <-ctx.Done():
+						return fmt.Errorf("re-sync cancelled via context: %s", ctx.Err())
+					}
+					_, _, err := IdentityAllocator.Allocate(ctx, globalIdentity{lbls})
```

Comment:
@tgraf is the `IdentityAllocator.Allocate` the right option here? I've a feeling the `IdentityAllocator.Allocate` has a ref counter inside and that's how `IdentityAllocator.Release` knows when an identity has its "lastUse"